src/client/push-client.js

```diff
@@ -175,24 +217,42 @@ export default class PushClient extends EventDispatch {
       }
     }
 
-    if (this.endpoint) {
+    if (this._endpoint) {
       // POST subscription details
-      this.endpoint.send({
+      this._endpoint.send({
         action: 'unsubscribe',
         subscription: subscription,
-        userId: this.userId
+        userId: this._userId
       });
     }
   }
 
+  /**
+   * If a service worker is registered for push, this method will return a
+   * promise that will resolve with the registration for that service worker.
+   *
+   * If there is no registration null will be the resolved reponse.
+   *
+   * @return {Promise<ServiceWorkerRegistration>} A Promise that
```

Comment:
If the method description says what is returned then I tend to skip the description for the return type here. It avoids writing the same thing twice.

So I would tend to get for either:

```javascript
/**
 * Get the registration of the service worker being used for push
 * @return {Promise<?ServiceWorkerRegistration>} A promise that resolves to either the current
 * registration, or to null if none.
 */
```

OR

```javascript
/**
 * Returns a promise that resolves to the registration of the service worker being used for push. If
 * there is no current registration the promise resolves to null
 * @return {Promise<?ServiceWorkerRegistration>}
 */
```